src/rpp/rpp/operators/zip.hpp

```diff
@@ -0,0 +1,129 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/combining_strategy.hpp>
+#include <rpp/operators/details/strategy.hpp>
+#include <rpp/schedulers/current_thread.hpp>
+
+#include <list>
+
+namespace rpp::operators::details
+{
+template<rpp::constraint::observer Observer, typename TSelector, rpp::constraint::decayed_type... Args>
+class zip_disposable final : public combining_disposable<Observer, TSelector, Args...>
+{
+public:
+    explicit zip_disposable(Observer&& observer, const TSelector& selector)
+        : combining_disposable<Observer, TSelector, Args...>(std::forward<Observer>(observer), selector)
+    {
+    }
+
+    auto& get_pendings() { return m_pendings; }
+
+private:
+    utils::tuple<std::list<Args>...> m_pendings{};
+};
+
+template<size_t I, rpp::constraint::observer Observer, typename TSelector, rpp::constraint::decayed_type... Args>
+struct zip_observer_strategy final
+    : public combining_observer_strategy<zip_disposable<Observer, TSelector, Args...>>
+{
+    using combining_observer_strategy<zip_disposable<Observer, TSelector, Args...>>::disposable;
+
+    template<typename T>
+    void on_next(T&& v) const
+    {
+        const auto observer = disposable->get_observer_under_lock();
+        disposable->get_pendings().template get<I>().push_back(std::forward<T>(v));
+
+        disposable->get_pendings().apply([this, &observer](auto&... values) {
+            if ((!values.empty() && ...))
+            {
+                observer->on_next(disposable->get_selector()(std::move(values.front())...));
+                (..., values.pop_front());
+            }
+        });
```

Comment:
This one can be extracted as separate function. Reason: reduce binary bloat: on_next function could be instantiated with const T&, T&, T&& and etc. For each of this one unique lambda would be instantiated

Reply:
Good point I didn't think about it, done